src/box/lua/call.c

```diff
@@ -1268,6 +1269,46 @@ lbox_func_new_or_delete(struct trigger *trigger, void *event)
 	return 0;
 }
 
+static int
+lbox_rt_access_reset(struct lua_State *L)
+{
+	if (lua_gettop(L) != 0)
+		return luaL_error(L, "internal.reset_lua_call takes no "
+				  "arguments, but %d given.", lua_gettop(L));
```

Comment:
I don't think we need to bother raising human-readable errors on invalid arguments for these functions because they're internal so it should be fine to use `luaL_checkstring` without checking `lua_top`, but if you do please:

1. Use box errors.
2. Change the error message to "Usage: ...".
3. Check argument types with `lua_type` and raise box errors for them as well.
4. Cover these checks by tests.

I mean something like this:

```C
if (lua_top(L) != 0) {
        diag_set(IllegalParams, "Usage: box.internal.lua_call_runtime_priv_reset()");
        return luaT_error(L);
}

if (lua_top(L) != 2 || lua_type(L, 1) != LUA_TSTRING || lua_type(L, 2) != LUA_TSTRING) {
        diag_set(IllegalParams, "Usage: box.internal.lua_call_runtime_priv_grant(user, func)");
        return luaT_error(L);
}
```

Reply:
Replaced with box errors. Added invalid usage test case.